adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRewardedAdLoader.m

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralRewardedAdLoader.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+
+#include <stdatomic.h>
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKReward/MTGRewardAd.h>
+#import <MTGSDKReward/MTGBidRewardAdManager.h>
+
+@interface GADMAdapterMintegralRewardedAdLoader ()<MTGRewardAdLoadDelegate,MTGRewardAdShowDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralRewardedAdLoader {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationRewardedLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// Ad configuration for the ad to be loaded.
+    GADMediationRewardedAdConfiguration *_adConfiguration;
+
+    /// The Mintegral rewarded ad.
+    MTGBidRewardAdManager *_rewardedAd;
+    
+    /// The ad event delegate to forward ad rendering events to the Google Mobile Ads SDK.
+    id<GADMediationRewardedAdEventDelegate> _adEventDelegate;
+}
+
+- (void)loadRewardedAdForAdConfiguration:(nonnull GADMediationRewardedAdConfiguration *)adConfiguration
+                       completionHandler:(nonnull GADMediationRewardedLoadCompletionHandler)completionHandler {
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationRewardedLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationRewardedAdEventDelegate>(
+                                                                        _Nullable id<GADMediationRewardedAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationRewardedAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
```

Comment:
Can we have `adUnitId` and `placementId` as iVars so we dont have to fetch them twice , and use it in `presentFromViewController:` ?

Reply:
OK, I've made changes and committed.